File: MMDSceneKitSample_macOS_ObjC/MMDSceneViewController.m
//
//  MMDSceneViewController.m
//  MMDSceneKitSample_macOS_ObjC
//
//  Created by magicien on 11/15/17.
//  Copyright © 2017 DarkHorse. All rights reserved.
//

#import "MMDSceneViewController.h"

@implementation MMDSceneViewController

- (void)setupGameScene:(SCNScene *)scene view:(MMDView *)view {
    @autoreleasepool {
        MMDSceneSource *source = [[MMDSceneSource alloc] initWithNamed:@"art.scnassets/サンプル（きしめんAllStar).pmm" options:nil models:nil];
        view.scene = [source getScene];
        view.showsStatistics = true;
        
        view.delegate = MMDIKController.sharedController;
        
        view.backgroundColor = NSColor.whiteColor;
    }
}

@end

File: Sample/Common/MMDSceneViewController.swift
//
//  MMDSceneViewController.swift
//  MMDSceneKit
//
//  Created by magicien on 2/13/16.
//  Copyright © 2016 DarkHorse. All rights reserved.
//

import SceneKit
#if os(OSX)
    import MMDSceneKit_macOS
    public typealias SuperViewController = NSViewController
    public typealias MMDView = SCNView
#elseif os(tvOS)
    import MMDSceneKit_tvOS
    public typealias SuperViewController = UIViewController
    public typealias MMDView = SCNView
#elseif os(iOS)
    import MMDSceneKit_iOS
    public typealias SuperViewController = UIViewController
    public typealias MMDView = SCNView
#elseif os(watchOS)
    import MMDSceneKit_watchOS
    public typealias SuperViewController = WKInterfaceController
    public typealias MMDView = WKInterfaceSCNScene
#endif

var mikuNode: MMDNode! = nil

public class MMDSceneViewController: SuperViewController {
    
    /**
     * setup game scene
     */
    public func setupGameScene(_ scene: SCNScene, view: MMDView) {
        // set the scene to the view
        view.scene = MMDSceneSource(named: "art.scnassets/サンプル（きしめんAllStar).pmm")!.getScene()!
        
        // show statistics such as fps and timing information
        view.showsStatistics = true
        
        #if !os(watchOS)
            // set the delegate to update IK for each frame
            view.delegate = MMDIKController.sharedController
        #endif
        
        // configure the view
        #if os(macOS)
            view.backgroundColor = NSColor.white
        #elseif os(iOS) || os(tvOS)
            view.backgroundColor = UIColor.white
        #endif
    }
}
